rfqmsg/buy_request.go

```diff
@@ -148,6 +162,17 @@ func (q *BuyRequest) Validate() error {
 			q.Version)
 	}
 
+	// Ensure that the suggested asset rate has not expired.
+	var err error
+	q.AssetRateHint.WhenSome(func(rate AssetRate) {
```

Comment:
nit: should use `fn.MapOptionZ` to return the error and not have the closure modify outside variables.